### PR TITLE
ingest: document index.default_pipeline

### PR DESCRIPTION
**Pull request by jakelandis**

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM

**Comment by jakelandis** [comment]
@original-brownbear - I forgot to call out this option in the ingest node documentation.  

Can you please take another look ? 
Specifically https://github.com/elastic/elasticsearch/pull/34500/commits/66f033cdc5450aef5113d841fa3f9074fb3f90a1 (I rendered the doc and the link works fine)

**Comment by original-brownbear** [comment]
Still LGTM :)

**Comment by rjernst** [comment]
Is the override of `pipeline=_none` mentioned somewhere already?

**Comment by jakelandis** [comment]
@rjernst 
> Is the override of pipeline=_none mentioned somewhere already?

Nope, it isn't mentioned. I can add to this PR.

how about:
```
 `index.default_pipeline`::
    The default <<ingest,ingest node>> pipeline for this index. Index requests will fail
    if the default pipeline is set and the pipeline does not exist. The default may be 
    be ignored per request by setting the request parameter `pipeline=_none`. 
```

**Comment by jakelandis** [comment]
Updated PR to include mention of `pipeline=_none`.

File: docs/reference/index-modules.asciidoc [hunk 1]
```diff
@@ -234,6 +234,12 @@ specific index module:
     The length of time that a <<delete-versioning,deleted document's version number>> remains available for <<index-versioning,further versioned operations>>.
     Defaults to `60s`.
 
+  `index.default_pipeline`::
+
+    The default <<ingest,ingest node>> pipeline for this index. Index requests will fail
+    if the default pipeline is set and the pipeline does not exist. The default may be
+    ignored per request by setting the request parameter `pipeline=_none`.
```
Comment (rjernst) [inline docs/reference/index-modules.asciidoc]: I suggestion something like this in place of the last sentence:

```
The default may be overridden using the `pipeline` parameter. The special pipeline name `_none` indicates no ingest pipeline should be run.
```

The point is to show that pipeline can be specified in general to override the default, and that `_none` is just a special pipeline name.

**Review by rjernst — APPROVED** [review]
Looks good, one suggestion.

**Comment by jakelandis** [comment]
6.5 backport: https://github.com/elastic/elasticsearch/commit/e0ee8022166a666899520ab6ec1e841940d353a6

**Pull request merged**